app/controllers/equipment_controller.rb

```diff
@@ -3,41 +3,43 @@
 require 'isbm_adaptor_common'
 
 class EquipmentController < ApplicationController
+  before_action :validate_source
 
   def new
+    # Alarms and Break Down Structures displays the same list of equipments.
+    # Hence, source here indicates the origin of this request.
+    # After the object gets saved, it gets redirected to its source based on this parameter.
     @equipment = Equipment.new
```

Comment:
Since things like UUIDs are quite complex and would often be autogenerated, I would either pre-populate the UUID with a generated value (`SecureRandom.uuid`) or have a button next to the field on the form that lets the user generate a value.